src/components/keypad/pre-algebra-page.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import React from "react";
+import {View} from "@khanacademy/wonder-blocks-core";
+import Color from "@khanacademy/wonder-blocks-color";
+
+import Keys from "../../data/key-configs";
+import {
+    KeypadPageContainer,
+    KeypadButton,
+    SecondaryKeypadButton,
+    KeypadActionButton,
+    PlaceHolderButton,
+} from "./keypad-page-items";
+
+import type {KeyConfig} from "../../data/key-configs";
+
+type Props = {|
+    onClickKey: (keyConfig: string) => void,
+|};
+type State = {};
+export default class NumericInputPage extends React.Component<Props, State> {
+    render() {
+        const {onClickKey} = this.props;
+        return (
+            <KeypadPageContainer>
+                {/* Row 1 */}
+                <SecondaryKeypadButton
+                    keyConfig={Keys.EXP_2}
+                    onClickKey={onClickKey}
+                />
+                <SecondaryKeypadButton
+                    keyConfig={Keys.EXP}
+                    onClickKey={onClickKey}
+                />
```

Comment:
A better approach to this might be to create 2D array with the keys and then generate the components from that, e.g.
```
const NO_KEY = null;
const keypadSpec = [
  [ Keys.EXP_2, Keys.ESP, Keys.SQRT, Keys.RADICAL, ... ],
  [ NO_KEY, NO_KEY, NO_KEY, Keys.X, ...],
  ...,
];

...

render() {
  return (
     <KeypadPageContainer>
        {keypadSpec.flatMap(row =>
            row.map(key => ... )
        )}
    </KeypadPageContainer>
  );
}
```